### DISCUSSION TITLE
How can i use `<script setup>` with vue3 template ?

### DISCUSSION THREAD

**Discussion by KeyToLove** [codesandbox/sandpack#647] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/54970106/204697777-1e022918-501f-419b-b528-7a30aef69885.png)

i had try like below，but it didn't work
![image](https://user-images.githubusercontent.com/54970106/204697623-6ba025ea-4d6d-4f6c-b45d-8a6cee015d88.png)

**Comment by danilowoz** [comment, marked as answer]
It has been fixed here: https://github.com/codesandbox/codesandbox-client/issues/6762 and here https://github.com/codesandbox/sandpack/pull/648

**Reply by KeyToLove** [reply]
oh wonderful 👍 !  one more question, if i don't use the default template (vue3),what should i do to support `<script setup>`.What are the necessary dependencies ?

**Reply by danilowoz** [reply]
You can pass your files and set the Vue environment, like this:

```js
<Sandpack
	files={files}
	customSetup={{ environment: "vue-cli" }}
/>
```

**Reply by KeyToLove** [reply]
got it~ Thanks for such a clear explanation :)
